tasks.py

```diff
@@ -0,0 +1,200 @@
+"""
+tasks.py defines tasks to run via ssh on one of the defined hosts.
+The functions with the @task decoration can be executed from the cli
+by calling `invoke task arg1 ... argn`, e.g. `invoke backup staging`.
+The caller has to be set up to make ssh connections to the intended host
+as the defined user.
+
+## Defined tasks
+* backup env
+    starting a routine to create an encrypted backup of the database
+    in the environment `env`.
+* deploy env github_user github_token
+    downloading the most recent image and code for the given environment,
+    then migrating the database if any migrations are necessary,
+    finally restarting the docker containers. The github user and token are
+    used to download the docker image from the github containe registry.
+* cleanup env
+    do cleanup tasks. At the moment this consists of removing unused docker
+    to free up space.
+
+
+"""
+import datetime
+import io
+import logging
+import os
+from dataclasses import dataclass
+
+from fabric import Connection
+from invoke import task
+
+
+logging.basicConfig(
+    format="[{asctime}] {levelname: <8} [{filename}:{lineno}] {message}",
+    level=logging.INFO,
+    style="{",
+)
+
+
+# make sure we are logging in UTC and have millisecond granularity
+def _formatTime(record, _):
+    return str(datetime.datetime.utcfromtimestamp(record.created)) + "Z"
+
+
+for h in logging.getLogger().handlers:
+    h.formatter.formatTime = _formatTime
+
+
+@dataclass
+class HostConfig:
+    """HostConfig collects various parameters for each deployment (prod, staging, etc)"""
+
+    host_name: str
+    git_branch: str
+    docker_version: str
+    github_user: str
+    github_token: str
+    backup_name: str
+    user: str = "deploy-web"
+    code_dir: str = "~/OpenOversight"
+    backup_dir: str = "~/openoversight_backup/"
+    image_path: str = "ghcr.io/lucyparsons/openoversight"
+    key_id: str = "OpenOversight"
+
+    def get_connection(self):
+        """Return ssh connection as used by fabric for this HostConfig."""
+        return Connection(host=self.host_name, user=self.user)
+
+
+def get_configs(
+    env: str, github_user: str = None, github_token: str = None
+) -> HostConfig:
+    """Return a HostConfig based on given environment and github user and token.
+    Github user and token can be omitted if no connection to github is needed.
+    """
```

Comment:
Nit:

```python
"""
Return a HostConfig based on given environment and github user and token.
Github user and token can be omitted if no connection to github is needed.
"""
```

Reply:
The same goes for the rest of the docstrings.

Reply:
Changed this one (because it is multi-line)